### PR TITLE
Emit Verneed

### PR DESCRIPTION
**Pull request by akawashiro**
Emit Verneed for each symbol in .dynsym.

CI fails although it works on my computer. Can I have a talk on Monday? I think the following line in `test-version.sh` does something wrong.
```
gcc -Wl,--hash-style=gnu -shared -Wl,-soname,libmax.so -Wl,--version-script,libmax2.def -o libmax.so.2 libmax2.o
```

### CODE REVIEW

File: elf_binary.cc [hunk 1]
```diff
@@ -295,20 +338,23 @@ Elf_Addr ELFBinary::OffsetFromAddr(Elf_Addr addr) {
...
+    std::stringstream ss;
     for (auto it : syms_) {
-        std::stringstream ss;
-        ss << it.first.second << ": " << it.first.first << " ";
-
-        if (versym_) {
-            ss << ShowVersym(it.first.second) << "\n";
+        ss << it.name << ": ";
+
+        if (it.ver == -1) {
```
Comment (shinh) [inline elf_binary.cc:345]: I feel now there are too much `if (x == VER_NDX_LOCAL) {} else if (x == VER_NDX_GLOBAL) {} else`. Could you consider factoring some of them out to helper functions?

Comment (akawashiro) [reply elf_binary.cc:345]: Thank you. I do it.

**Comment by akawashiro** [comment]
This PR includes some of [PR17](https://github.com/shinh/sold/pull/17). So, it may cause a conflict.

**Comment by akawashiro** [comment]
I confirmed this PR works both of ubuntu 18.04 and ubuntu 20.04.

**Pull request merged**